libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -91,6 +91,12 @@ void NavEKF3_core::setWindMagStateLearningMode()
         manoeuvring = false;
     }
 
+    if (finalInflightYawInit && !finalInflightMagInit) {
```

Comment:
Better to fix the root cause which I understand to be the failure to set magFieldLearned to false when on ground. When I reverted this patch and added one to reset magFieldLearned when on ground, the second takeoff was doing mag fusion.